src/crypto_tools/k256_serde.rs

```diff
@@ -84,31 +82,88 @@ impl<'de> Deserialize<'de> for Scalar {
     }
 }
 
+/// SingingKey does not impl Zeroize and so cannot have #[zeroize(drop)] so we need to roll our own
+#[derive(Clone, Debug, PartialEq)]
+pub struct SigningKey(k256::ecdsa::SigningKey);
+
+impl AsRef<k256::ecdsa::SigningKey> for SigningKey {
+    fn as_ref(&self) -> &k256::ecdsa::SigningKey {
+        &self.0
+    }
+}
+
+impl From<k256::ecdsa::SigningKey> for SigningKey {
+    fn from(s: k256::ecdsa::SigningKey) -> Self {
+        SigningKey(s)
+    }
+}
+
+/// SingingKey does not impl Zeroize so we need to roll our own
+impl Zeroize for SigningKey {
+    fn zeroize(&mut self) {
+        // SigningKey is a NonZeroScalar under the hood
+        // NonZeroScalar impls Zeroize, so use that
+        let zeroizable = unsafe {
+            &mut *(&mut self.0 as *mut k256::ecdsa::SigningKey as *mut k256::NonZeroScalar)
```

Comment:
Yikes. Might be worth pushing a PR to add `Zeroize` impl for `SigningKey` in `k256`?